### PR TITLE
[Private DC] Support API for /place/stat-vars, /v1/place/stat-vars/union

### PR DESCRIPTION
**Pull request by shifucun**
This is used by scatter tool.

Also fixed stat var path bug for private stat vars

### CODE REVIEW

File: internal/server/place_stat_vars.go [hunk 1]
```diff
@@ -99,53 +105,72 @@ func keysToSlice(m map[string]bool) []string {
...
+				if _, ok := filterStatVarSet[sv]; ok {
+					set[sv] = true
+				}
+			}
 		}
 	}
-	return &pb.GetPlaceStatVarsUnionResponse{
-		StatVars: keysToSlice(set),
-	}, nil
+	result.StatVars = util.MergeDedupe(result.StatVars, keysToSlice(set))
```
Comment (pradh) [inline internal/server/place_stat_vars.go:174]: Can this MergeDedupe be inside the memDb check?

File: internal/server/place_stat_vars.go [hunk 2]
```diff
@@ -99,53 +105,72 @@ func keysToSlice(m map[string]bool) []string {
...
+) (*pb.GetPlaceStatVarsUnionResponse, error) {
+	// Check places
+	places := in.GetDcids()
+	if len(places) == 0 {
 		return nil, status.Error(
 			codes.InvalidArgument, "Missing required arguments: dcids")
 	}
+	// filtered stat vars
+	filterStatVars := in.GetStatVars()
+	filterStatVarSet := map[string]struct{}{}
```
Comment (pradh) [inline internal/server/place_stat_vars.go:118]: Is this fixing a bug from before?

maybe add comment like "// Stat vars may have dups, so compute set."

**Review by pradh — APPROVED** [review]

File: internal/server/place_stat_vars.go (hunk 1 above)
Comment (shifucun) [reply internal/server/place_stat_vars.go]: Good point!

File: internal/server/place_stat_vars.go (hunk 2 above)
Comment (shifucun) [reply internal/server/place_stat_vars.go:119]: This is more of checking the stat vars efficiently with a map. Added a comment.

**Review by shifucun — COMMENTED** [review]
Thanks for review!

**Pull request merged**